### DISCUSSION TITLE
How to import civet from ts?

### DISCUSSION THREAD

**Discussion by NWYLZW** [DanielXMoore/Civet#1499] [Q&A] [answered]
- `foo.civet`
```civet
export foo := 'foo'
```
- `index.ts`
```ts
import { foo } from './foo.civet'
```

**Comment by edemaine** [comment]
That is generally the way, but of course you need to compile/build. Are you using a build tool?

**Comment by NWYLZW** [comment]
I'm sorry. I didn't describe the problem clearly. Actually, what I want to ask is how to make this import have the corresponding types in the IDE. I don't have any problems with the build.
<img width="691" alt="image" src="https://github.com/user-attachments/assets/e60414e4-81e1-43dd-85c5-da57bd1eb684">
Do I have to compile the file contents to TypeScript to get the corresponding types? Is it possible to make a TypeScript plugin to support it?

**Comment by edemaine** [comment, marked as answer]
[Civet's VSCode plugin](https://marketplace.visualstudio.com/items?itemName=DanielX.civet) should already support this case. If you haven't installed it, please do. Then, you may also need to disable VSCode's built-in TypeScript plugin, because Civet's should handle everything for you.

**Reply by NWYLZW** [reply]
Oh no. It seems that it just got stuck for a moment.
~~I don't know why after I configured the ts file and also processed it with Civet's LSP, neither the type checking nor the code hinting seems to be working.~~

<img width="1492" alt="image" src="https://github.com/user-attachments/assets/0be9ffc6-e1ef-412b-bd75-1bf2fd801c2e">

<img width="300" alt="image" src="https://github.com/user-attachments/assets/e2efa228-e3bd-4ade-b986-c5b6e562d932">

<img width="616" alt="image" src="https://github.com/user-attachments/assets/518bdc63-a072-49ef-b2da-b338c532ca17">
